File: typescript/src/models/eventDataOrder.ts
// this file is @generated
import { type Refund, RefundSerializer } from "./refund";
import {
  type WebhookOrderStatus,
  WebhookOrderStatusSerializer,
} from "./webhookOrderStatus";

export interface EventDataOrder {
  type: string;
  checkoutSessionId: string;
  permalinkUrl: string;
  status: WebhookOrderStatus;
  refunds: Refund[];
}

export const EventDataOrderSerializer = {
  _fromJsonObject(object: any): EventDataOrder {
    return {
      type: object["type"],
      checkoutSessionId: object["checkout_session_id"],
      permalinkUrl: object["permalink_url"],
      status: WebhookOrderStatusSerializer._fromJsonObject(object["status"]),
      refunds: object["refunds"].map((item: Refund) =>
        RefundSerializer._fromJsonObject(item)
      ),
    };
  },

  _toJsonObject(self: EventDataOrder): any {
    return {
      type: self.type,
      checkout_session_id: self.checkoutSessionId,
      permalink_url: self.permalinkUrl,
      status: WebhookOrderStatusSerializer._toJsonObject(self.status),
      refunds: self.refunds.map((item) => RefundSerializer._toJsonObject(item)),
    };
  },
};

File: rust/src/models/event_data_order.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{refund::Refund, webhook_order_status::WebhookOrderStatus};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct EventDataOrder {
    pub r#type: String,
    pub checkout_session_id: String,
    pub permalink_url: String,
    pub status: WebhookOrderStatus,
    pub refunds: Vec<Refund>,
}
